Fixed failing OIDC test issues

blen-desta opened the pull request:
- Fixed failing OIDC test issues
- Added OIDC configuration defaults to blueprint
- Added OIDC logout and credential flow tests
- Added error tests
- Added ID token validation on login
- Modified the request to the userinfo endpoint to be a GET

blen-desta committed:
- Fixed failing OIDC test issues
- Added OIDC configuration defaults to blueprint
- Added OIDC logout and credential flow tests
- Added error tests
- Added ID token validation on login
- Modified the request to the userinfo endpoint to be a GET

bakejeyner commented on platform/security/handler/security-handler-oidc/src/main/java/org/codice/ddf/security/handler/oidc/OidcHandler.java:
```diff
@@ -191,6 +216,13 @@ private boolean userAgentIsNotBrowser(HttpServletRequest httpRequest) {
+        && !requestParams.containsKey("access_token")
+        && !requestParams.containsKey("id_token")) {
+      return new OidcCredentials();
+    }
```
❗️ I believe it is possible to receive an access token in the authentication header https://auth0.com/docs/tokens/guides/access-token/use-access-tokens#custom-api-access-token-interactions.

bakejeyner commented on platform/security/handler/security-handler-oidc/src/main/resources/OSGI-INF/blueprint/blueprint.xml:
```diff
@@ -14,29 +14,38 @@
-
     <bean id="oidcConfiguration" class="org.codice.ddf.security.handler.oidc.OidcHandlerConfigurationImpl">
         <cm:managed-properties persistent-id="org.codice.ddf.security.handler.api.OidcHandlerConfiguration"
                                update-strategy="component-managed" update-method="update"/>
```
❗️ Should the `update-strategy`/`update-method` here be changed? I know before blueprint would call the update method with an updated `Map<String,Object>` properties object, that we would copy to our own properties object.

Heres a reference on what `update-strategy` to chose, summarized from [here](https://wiki.opendaylight.org/view/Using_Blueprint):
```
None: doesn't update the blueprint container
Reload: dynamically reload the blueprint container and re-create all the beans
Component-Managed: updates to the properties are essentially handled by the component, ie when the .cfg file is changed, the method specified via the update-method attribute is invoked with the Map of properties
Container-Managed: the container invokes setter methods on the bean corresponding to the configuration property keys. update-method is not used with this strategy.
```
```

SmithJosh committed:
Fix session management endpoint

This required moving the SecurityAssertion implementations out of the
security-core-commons dependency. Since that dependency gets embedded,
it wasn't possbile for the session management service to check the
implementation of a given SecurityAssertion (the assertion gets created
in a different bundle).

SmithJosh committed:
Move security-servlet-logout to security-logout

SmithJosh committed:
Replace logout UI component with single logout page

SmithJosh committed:
Update security-servlet-logout and session-management-impl unit tests

blen-desta committed:
- Fixed userinfo token validating
- Added CustomOidcProfileCreator which creates an OidcProfile in the OidcRealm and adds it to the SecurityAssertionJwt so the OidcLogoutActionProvider doesn't have to create one

SmithJosh committed:
Fix context policy manager whitelist entry for favicon

blen-desta committed:
More modifiations

blen-desta commented:
Changes pushed up to https://github.com/stustison/ddf/tree/oidc-rebase
